### ISSUE TITLE
Failure to wake from suspend on kernel 6.11.5

### ISSUE THREAD

**Issue by Jezurko** [getsolus/packages#4233]
### Please confirm there isn't an existing open bug report

- [X] I have searched open bugs for this issue

### Summary

Hello,
since updating to kernel 6.11.5 I am unable to wake my computer from suspend.
This happens every time I try to suspend the computer, it doesn't seem to matter for how long it is suspended.
After booting in the previous kernel (6.10) I can put the machine to sleep and wake up without issues.

### Steps to reproduce

1. Suspend the computer from user interface
2.  Try to wake up the computer

### Expected result

The computer succesfully wakes up and can be used

### Actual result

The computer stays frozen, doesn't output any image, doesn't react to inputs, doesn't show up on the network (can't ssh into it.)

### Environment

- [X] Is system up to date?

### Repo

Shannon (stable)

### Desktop Environment

Budgie

### System details

System:
  Host: woody Kernel: 6.11.5-307.current arch: x86_64 bits: 64
  Desktop: Budgie v: 10.9.2 Distro: Solus 4.6 convergence
Machine:
  Type: Desktop System: ASUS product: N/A v: N/A serial: <superuser required>
  Mobo: ASUSTeK model: TUF GAMING X670E-PLUS WIFI v: Rev 1.xx
    serial: <superuser required> UEFI: American Megatrends v: 3014
    date: 07/17/2024
CPU:
  Info: 8-core AMD Ryzen 7 7800X3D [MT MCP] speed (MHz): avg: 5040
    min/max: 545/5050
Graphics:
  Device-1: Advanced Micro Devices [AMD/ATI] Navi 31 [Radeon RX 7900 XT/7900
    XTX/7900 GRE/7900M] driver: amdgpu v: kernel
  Device-2: Advanced Micro Devices [AMD/ATI] Raphael driver: amdgpu
    v: kernel
  Display: x11 server: X.Org v: 21.1.14 with: Xwayland v: 24.1.4 driver: X:
    loaded: amdgpu unloaded: fbdev,modesetting,radeon,vesa dri: radeonsi
    gpu: amdgpu resolution: 3840x2160
  API: OpenGL v: 4.6 compat-v: 4.5 vendor: amd mesa v: 24.2.5 renderer: AMD
    Radeon RX 7900 XT (radeonsi navi31 LLVM 18.1.8 DRM 3.59
    6.11.5-307.current)
Network:
  Device-1: Realtek RTL8125 2.5GbE driver: r8169
  Device-2: MEDIATEK MT7921K Wi-Fi 6E 80MHz driver: mt7921e
Drives:
  Local Storage: total: 2.74 TiB used: 1.41 TiB (51.6%)
Info:
  Memory: total: 64 GiB available: 61.94 GiB used: 2.36 GiB (3.8%)
  Processes: 426 Uptime: 0m Shell: Bash inxi: 3.3.36

### Other comments

I have tried using HDMI cable instead of DP cable to conenct my monitor, no change.
I have a NFS share and SAMBA share mounted on boot. I tried unmounting those before suspend, no change.
Only succesful remedy was booting into the 6.10 kernel

The systemd log shows no information about the attempted wake up. Attaching log about the suspend
[journal.log](https://github.com/user-attachments/files/17611949/journal.log)

**Comment by OlivierM16** [comment]
I have the same problem with a fresh install. Network Manager seems to be involved in this issue.

[log_previous_boot_sudo.txt](https://github.com/user-attachments/files/17778505/log_previous_boot_sudo.txt)

**Comment by Jezurko** [comment]
Curiously enough, after the last sync, my computer seems to be able to wake up again. I don't think I have changed anything relevant on my system, so it was probably something in the update.

**Comment by Jezurko** [comment]
Okay, I was too quick with my assessment, the bug is still present. But I finally found the culprit. 
TLDR: Mediatek mt7921e driver has some bugs.

I have a wireless card MT7921K (RZ608) from Mediatek and apparently it's a known troublemaker…
I found [this arch forum thread](https://bbs.archlinux.org/viewtopic.php?id=299987&p=3) about the same issue.
There are few issues for the driver of this wireless card on the [kernel bugzilla](https://bugzilla.kernel.org/buglist.cgi?quicksearch=mt7921e), some of them seem relevant to my issue.

I added the kernel parameter (`mt7921e.disable_aspm=Y`) to my boot options - that on its own doesn't help.

What works for me is to disable bluetooth before suspend (I have wifi disabled all the time at this machine - maybe it's necessary to also turn that off). After that I can resume from suspend. If I have bluetooth enabled before suspend, it refuses to wake-up.

This part isn't exactly important for the bug, but it explains why I thought the bug was fixed:
Apparently this wireless card also has a random issue, where it stops working. After I noticed that my bluetooth stopped working (and that must have happened the same moment my sleep started working) I stumbled upon [this post from endevaour os forum](https://forum.endeavouros.com/t/mediatek-mt7922-bluetooth-not-recognized/39841/16) that recommended to power off the machine and disconnect the power cord for a moment + press the power button. This restored my bluetooth functionality, but made the sleep issue reappear.

**Comment by TraceyC77** [comment]
Closing due to age, the kernel having been updated, and that we aren't the maintainers of the Mediatek kernel drivers.
